Core/Utilities/TimeLeft/MJFTimeLeft.py

```diff
@@ -0,0 +1,93 @@
+########################################################################
+# $Id$
+########################################################################
+
+""" The Machine/Job Features TimeLeft utility interrogates the MJF values
+    for the current CPU and Wallclock consumed, as well as their limits.
+"""
+
+from DIRAC import gLogger, S_OK, S_ERROR
+
+__RCSID__ = "$Id$"
+
+import os, re, time, urllib
+
+class MJFTimeLeft:
+
+  #############################################################################
+  def __init__( self ):
+    """ Standard constructor
+    """
+    self.log = gLogger.getSubLogger( 'MJFTimeLeft' )
+    self.jobID = None
+    if os.environ.has_key( 'JOB_ID' ):
+      self.jobID = os.environ['JOB_ID']
+    self.queue = None
+    if os.environ.has_key( 'QUEUE' ):
+      self.queue = os.environ['QUEUE']
+
+    self.cpuLimit = None
+    self.wallClockLimit = None
+    self.log.verbose( 'jobID=%s, queue=%s' % ( self.jobID, self.queue ) )
+    self.startTime = time.time()
+
+  #############################################################################
+  def getResourceUsage( self ):
+    """Returns a dictionary containing CPUConsumed, CPULimit, WallClockConsumed
+       and WallClockLimit for current slot.  All values returned in seconds.
+    """
+
+    cpu = None
+    cpuLimit = None
+    wallClock = None
+    wallClockLimit = None
+
+    try:
+      # We are not called from TimeLeft.py if these are not set
+      jobFeaturesPath = os.environ['JOBFEATURES']
+      machineFeaturesPath = os.environ['MACHINEFEATURES']
+    except:
+      self.log.warn( '$JOBFEATURES and $MACHINEFEATURES not set' )
+
+    try:
+      wallClockLimit = int( urllib.urlopen(jobFeaturesPath + '/wall_limit_secs').read() )
```

Comment:
Is this limiting the usage of this plugin to cases when the files are on a web server? On batch systems that are a regular file, e.g. /etc/machinefeatures
We need a versatile code that can read these files both locally and as a URL